gamemodes/terrortown/gamemode/server/sv_voice.lua

```diff
@@ -119,41 +122,65 @@ local function SendRoleVoiceState(speaker)
 	end
 end
 
-local function RoleGlobalVoice(ply, cmd, args)
+local function RoleGlobalVoice(ply, state)
 	if not IsValid(ply) or not ply:IsActive() or ply:GetSubRoleData().unknownTeam or ply:GetSubRoleData().disabledTeamVoice then return end
 
 	local tm = ply:GetTeam()
 	if tm == TEAM_NONE or TEAMS[tm].alone then return end
 
-	if #args ~= 1 then return end
+	ply[tm .. "_gvoice"] = state
 
-	local state = tonumber(args[1])
+	SendRoleVoiceState(ply)
+end
 
-	ply[tm .. "_gvoice"] = state == 1
+local function NetRoleGlobalVoice(len, ply)
+	local state = net.ReadBool()
 
-	SendRoleVoiceState(ply)
+	RoleGlobalVoice(ply, state)
+end
+net.Receive("TTT2RoleGlobalVoice", NetRoleGlobalVoice)
+
+local function ConCommandRoleGlobalVoice(ply, cmd, args)
+	if #args ~= 1 then return end
+
+	local state = tonumber(args[1]) == 1
+
+	RoleGlobalVoice(ply, state)
 end
-concommand.Add("tvog", RoleGlobalVoice)
+concommand.Add("tvog", ConCommandRoleGlobalVoice)
 
-local function MuteTeam(ply, cmd, args)
-	if not IsValid(ply) or not #args == 1 and tonumber(args[1]) then return end
+local function MuteTeam(ply, state)
+	if not IsValid(ply) then return end
 
 	if not ply:IsSpec() then
 		ply.mute_team = -1
 
 		return
 	end
 
-	local t = tonumber(args[1])
-
-	ply.mute_team = t
+	ply.mute_team = state
 
-	if t == MUTE_ALL then
+	if state == MUTE_ALL then
 		ply:ChatPrint("All muted.")
-	elseif t == MUTE_NONE or t == TEAM_UNASSIGNED or not team.Valid(t) then
+	elseif state == MUTE_NONE or state == TEAM_UNASSIGNED or not team.Valid(state) then
 		ply:ChatPrint("None muted.")
 	else
-		ply:ChatPrint(team.GetName(t) .. " muted.")
+		ply:ChatPrint(team.GetName(state) .. " muted.")
 	end
 end
-concommand.Add("ttt_mute_team", MuteTeam)
+
+local function NetMuteTeam(len, ply)
+	local state = net.ReadBool()
+
+	MuteTeam(ply, state)
+end
+net.Receive("TTT2MuteTeam", NetMuteTeam)
+
+local function ConCommandMuteTeam(ply, cmd, args)
+	if not #args == 1 and tonumber(args[1]) then return end
```

Comment:
`if not x == y` haha. Better use `~=` :)

Reply:
especially since `not a == b` inverts a first before comparing it to b, this isn't a problem for booleans, but can be a problem in other scenarios

Reply:
This was an artifact. Done